lib/simplifyExpression/basicsSearch/reduceMultiplicationByZero.js

```diff
@@ -1,27 +1,16 @@
+const {build, query} = require('math-nodes');
+
 const ChangeTypes = require('../../ChangeTypes');
 const Node = require('../../node');
 
 // If `node` is a multiplication node with 0 as one of its operands,
 // reduce the node to 0. Returns a Node.Status object.
 function reduceMultiplicationByZero(node) {
-  if (node.op !== '*') {
-    return Node.Status.noChange(node);
-  }
-  const zeroIndex = node.args.findIndex(arg => {
-    if (Node.Type.isConstant(arg) && arg.value === '0') {
-      return true;
-    }
-    if (Node.PolynomialTerm.isPolynomialTerm(arg)) {
-      const polyTerm = new Node.PolynomialTerm(arg);
-      return polyTerm.getCoeffValue() === 0;
-    }
-    return false;
-  });
-  if (zeroIndex >= 0) {
-    // reduce to just the 0 node
-    const newNode = Node.Creator.constant(0);
+  // The math-rules rule for this didn't do the right thing when the '0' was
+  // in the middle of the args list.
```

Comment:
I suppose we could run one and then run the other one again, but it still doesn't help with situations like `a * b * c * 0 * e * f * g` b/c running each once would result in `a * b * 0 * f * g` so we'd have to run them multiple times and I believe the intent of reduceMultiplicationByZero is that we transform a product with at least one factor that is `0` to `0`.